plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseConnectorTest.java

```diff
@@ -674,6 +674,25 @@ protected TestTable simpleTable()
         return new TestTable(onRemoteDatabase(), "tpch.simple_table", "(col BIGINT) Engine=Log", ImmutableList.of("1", "2"));
     }
 
+    @Override
+    public void testCreateTableWithLongTableName()
+    {
+        // Override because ClickHouse connector can create a table which can't be dropped
+        String baseTableName = "test_create_" + randomTableSuffix();
+        String validTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length());
+
+        assertUpdate("CREATE TABLE " + validTableName + " (a bigint)");
+        assertTrue(getQueryRunner().tableExists(getSession(), validTableName));
+        assertThatThrownBy(() -> assertUpdate("DROP TABLE " + validTableName))
+                .hasMessageMatching("(?s).*(Bad path syntax|File name too long).*");
+
+        String invalidTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length() + 1);
+        assertThatThrownBy(() -> query("CREATE TABLE " + invalidTableName + " (a bigint)"))
+                .hasMessageMatching("(?s).*(Cannot open file|File name too long).*");
+        // ClickHouse lefts a table even if the above statement failed
+        assertTrue(getQueryRunner().tableExists(getSession(), validTableName));
+    }
```

Comment:
add
```
assertThatThrownBy(() -> query("DROP TABLE ...
```

so that we know to update the test when clickhouse behavior is fixed

Reply:
The logic to capture future ClickHouse change already exists in L686.